releasenotes/notes/expose-qk-complex64-3b0a134669b2def4.yaml

```diff
@@ -0,0 +1,41 @@
+---
+upgrade_c:
+  - |
+    The way complex numbers are exposed in Qiskit's C API has been updgraded.
+    Previously, ``QkComplex64`` was a compiler-dependent typedef that allowed to pass
+    native complex types by pointer to Qiskit's API (for example as ``double complex*``).
+    While this was convenient, this approach implictly relied on memory-layout assumptions
+    that are not strictly guaranteed.
+
+    Qiskit v2.1 now exposes ``QkComplex64 { double re; double im; }`` as struct, to ensure the
+    memory layout is always compatible and for broader compiler support. For convenience,
+    compiler-dependent converters ``qk_complex64_from_native`` and ``qk_complex64_to_native``
+    are provided, which allows translating from the struct to a native complex number.
+    Note that these only work on platforms supporting ``double complex`` or for MSVC compilers
+    using ``_Dcomplex``.
+
+    For example::
+
+      #include <qiskit.h>
+      #include <math.h>
+      #include <stdio.h>
+      #include <complex.h>
+
+      void main() {
+          // using converter
+          double complex native = 5.0 + I * 3; // uses C11 standard, does not work on MSVC
+          QkComplex64 coeff = qk_complex64_from_native(&native);  // convert from native
+
+          // platform-independent constructions:
+          // QkComplex64 coeff = {5.0, 3.0};
```

Comment:
Why do we put the `double complex native` line in regular text, but the `QkComplex64 coeff` one in a comment?  The second one will always work.

Reply:
Yeah fair point, it's also shorter lol